build/index.js

```diff
@@ -0,0 +1,119 @@
+
+// Adapted from the countup.js format number function
+// https://github.com/inorganik/countUp.js/blob/master/countUp.js#L46-L60
+var formatNumber = function formatNumber(start, options) {
+  var num = '' + start.toFixed(options.decimals);
+  var x = num.split('.');
+  var x1 = x[0];
+  var x2 = x.length > 1 ? '' + options.decimal + x[1] : '';
+  var rgx = /(\d+)(\d{3})/;
+
+  if (options.useGrouping && options.separator) {
+    while (rgx.test(x1)) {
+      x1 = x1.replace(rgx, '$1' + options.separator + '$2');
+    }
+  }
+  return '' + options.prefix + x1 + x2 + options.suffix;
+};
+
+var startAnimation = function startAnimation(component) {
+  if (!(component && component.spanElement)) {
+    throw new Error('You need to pass the CountUp component as an argument!\neg. this.myCountUp.startAnimation(this.myCountUp);');
+  }
+
+  var countupInstance = new _countup2.default(component.spanElement, component.props.start, component.props.end, component.props.decimals, component.props.duration);
+
+  countupInstance.start();
+};
+
+var CountUp = function (_Component) {
+  _inherits(CountUp, _Component);
+
+  function CountUp() {
+    _classCallCheck(this, CountUp);
+
+    return _possibleConstructorReturn(this, (CountUp.__proto__ || Object.getPrototypeOf(CountUp)).apply(this, arguments));
+  }
+
+  _createClass(CountUp, [{
+    key: 'componentDidMount',
+    value: function componentDidMount() {
+      startAnimation(this);
+    }
+  }, {
+    key: 'shouldComponentUpdate',
+    value: function shouldComponentUpdate(nextProps) {
+      var hasCertainPropsChanged = this.props.duration !== nextProps.duration || this.props.end !== nextProps.end || this.props.start !== nextProps.start;
+
+      return nextProps.redraw || hasCertainPropsChanged;
+    }
+  }, {
+    key: 'componentDidUpdate',
+    value: function componentDidUpdate() {
+      startAnimation(this);
+    }
+  }, {
+    key: 'refSpan',
+    value: function refSpan(span) {
+      this.spanElement = span;
+    }
+  }, {
+    key: 'render',
+    value: function render() {
+      var _this2 = this;
+
+      return _react2.default.createElement(
+        'span',
+        {
+          className: this.props.className,
+          ref: function ref(span) {
+            _this2.refSpan(span);
+          }
+        },
+        formatNumber(this.props.start, {
+          decimal: this.props.decimal,
+          decimals: this.props.decimals
+        })
+      );
+    }
+  }]);
+
+  return CountUp;
+}(_react.Component);
+
+exports.default = CountUp;
+
+
+CountUp.propTypes = {
+  className: _react.PropTypes.string,
+  decimal: _react.PropTypes.string,
+  decimals: _react.PropTypes.number,
+  duration: _react.PropTypes.number,
+  easingFn: _react.PropTypes.func,
+  end: _react.PropTypes.number,
+  redraw: _react.PropTypes.bool, // eslint-disable-line react/no-unused-prop-types
+  start: _react.PropTypes.number
+};
```

Comment:
This whole file should likely be .gitignore'd, no?

Reply:
I've been trying to get this to work with gitignoring this file but the "prepare" npm script isn't working with a local install like it is from the actual npm module. I worked with Abiodun a bit to try and find a different script we could use or any other way to build this build folder locally but was unsuccessful.